pkg/bff/gds_test.go

```diff
@@ -0,0 +1,56 @@
+package bff_test
+
+import (
+	"context"
+
+	"github.com/trisacrypto/directory/pkg/bff/api/v1"
+	"github.com/trisacrypto/directory/pkg/bff/mock"
+	"google.golang.org/grpc/codes"
+)
+
+func (s *bffTestSuite) TestLookup() {
+	require := s.Require()
+	params := &api.LookupParams{}
+
+	// Test Bad Request (no parameters)
+	_, err := s.client.Lookup(context.TODO(), params)
+	require.EqualError(err, "[400] must provide either uuid or common_name in query params", "expected a 400 error with no params")
+
+	// Provide some params
+	params.CommonName = "api.alice.vaspbot.net"
+
+	// Test NotFound returns a 404
+	require.NoError(s.testnet.UseError(mock.LookupRPC, codes.NotFound, "not found"))
+	require.NoError(s.mainnet.UseError(mock.LookupRPC, codes.NotFound, "not found"))
+	_, err = s.client.Lookup(context.TODO(), params)
+	require.EqualError(err, "[404] no results returned for query", "expected a 404 error when both GDS return not found")
```

Comment:
GDSes maybe? The pluralization sounds a bit weird but makes more sense grammatically I think